test/fn.spec.php

```diff
@@ -381,6 +381,36 @@ function() { yield from [1,2,3]; }
             $res = index('a', ['b' => 1], 2);
             expect($res)->equal(2);
         });
+        test('Also works with objects that implement ArrayAccess', function() {
+            class MyClass implements \ArrayAccess
```

Comment:
I like this, any chance you could use an `ArrayObject` instead of defining our own class right here?

Reply:
I considered it (since defining the class takes many lines) but `array_key_exists` actually works with `ArrayObject` instances; so I figured I'd use something that failed if it reached `array_key_exists` by mistake

Reply:
Ah, great to know!